FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -810,17 +818,19 @@ ContextImpl::CompileCodeResult ContextImpl::CompileCode(FEXCore::Core::InternalT
   };
 }
 
-uintptr_t ContextImpl::CompileBlock(FEXCore::Core::CpuStateFrame* Frame, uint64_t GuestRIP, uint64_t MaxInst) {
+uintptr_t ContextImpl::CompileBlock(FEXCore::Core::CpuStateFrame* Frame, uint64_t GuestRIP, uint64_t MaxInst, bool SkipCache) {
```

Comment:
Does executing the BlockJITNaming branch make any sense with SkipCache? Similarly, does it make sense to include single-instruction compiles in profiles?

Truly the EFLAGS.TF path only needs to acquire the lock and call CompileCode directly. Semantically, a "SkipCache" parameter doesn't model this: The intent is never to "skip the cache" but to enable single-stepping, so implementing one via the other is a leaky abstraction.

Instead of hiding 90% of this function in an ad-hoc branch, I'd opt for a dedicated interface that's just called something like "PrepareSingleStep". Just inlining the required logic where needed would also work.